components/mpas-ocean/src/shared/mpas_ocn_gm.F

```diff
@@ -647,6 +649,69 @@ subroutine ocn_GM_compute_Bolus_velocity(statePool, &
            !$omp end parallel
          end if
 
+         ! When config_GM_closure = 'Griffies' actually modify the value of gmBolusKappa based on
+         ! Griffies et al (2005)
+         if (local_config_GM_compute_Griffies) then
+
+            ! This is Griffies et al. (2005) eqn 15:
+            ! A_{gm} = alpha |grad_z rho| L^2 g / ( rho_0 N_0 )
+            ! here
+            ! coef = alpha L^2 g / ( rho_0 N_0 )
+            ! A_{gm} = coef * |grad_z rho|
+            coef = config_GM_Griffies_alpha * config_gm_Griffies_length**2 * gravity &
+                   / (config_GM_Griffies_ref_density*config_GM_Griffies_No + 1.0E-15)
+
+            !$omp parallel
+            !$omp do schedule(runtime) private(k, cell1, cell2, sum_gradDensity, ltSum, countN2, &
+            !$omp             zEdge, ltAvg)
+            do iEdge = 1, nEdges
+               k=minLevelEdgeBot(iEdge)
+               sum_gradDensity = 0.0_RKIND
+               ltSum = 0.0_RKIND
+               countN2 = 0
+               cell1 = cellsOnEdge(1, iEdge)
+               cell2 = cellsOnEdge(2, iEdge)
+
+               k = minLevelEdgeBot(iEdge)+1
+               zEdge = -layerThickEdgeMean(k-1,iEdge)
+               do while(zEdge > -config_GM_Griffies_min_depth .and. k < maxLevelEdgeTop(iEdge))
+                  k = k+1
+                  zEdge = zEdge - layerThickEdgeMean(k-1,iEdge)
+               end do
+
+               do while(zEdge > -config_GM_Griffies_max_depth .and. k < maxLevelEdgeTop(iEdge))
```

Comment:
Performance note: This is for a general vertical coordinate, but we run all standard simulations with z-star (except under ice shelves, where GM doesn't really matter). We could have two versions of this - one that computes (say) `k_GM_Griffies_start` and `k_GM_Griffies_end` on init, and another that computes it for every edge on every time step, chosen with a flag between the two.